implementations/rust/ockam/ockam_transport_tcp/src/portal/portal_message.rs

```diff
@@ -161,8 +162,7 @@ mod test {
         let decoded = PortalMessageV1::decode(&encoded).unwrap();
         assert!(matches!(decoded, PortalMessageV1::Disconnect));
 
-        let encoded =
-            PortalMessage::encode(PortalMessage::Payload(payload.clone(), Some(123))).unwrap();
+        let encoded = PortalMessage::encode(PortalMessage::Payload(&payload, Some(123))).unwrap();
```

Comment:
Do you think we should have test vectors here and in other places where you used manual encoding?

Reply:
Where manually encoding is used i've added tests to check compatibility with serde own serialization, i think it's enough to cover sane cases (if serde own encoding changes, everything would crumple). On the long run we might want to start some schema of sort.